MapboxSearch/sdk/src/main/java/com/mapbox/search/SearchEngineImpl.kt

```diff
@@ -306,6 +310,49 @@ internal class SearchEngineImpl(
         }
     }
 
+    override fun retrieve(mapboxId: String, executor: Executor, callback: SearchResultCallback): AsyncOperationTask {
+        val requestOptions = RequestOptions(
+            "",
+            "",
+            EMPTY_SEARCH_OPTIONS,
+            false,
+            false,
+            NO_SESSION_IDENTIFIER,
+        )
+        val suggestAction = when (this.apiType) {
+            ApiType.SBS -> {
+                SuggestAction(
+                    "retrieve",
+                    "",
+                    null,
+                    """{"id":"$mapboxId"}""".toByteArray(),
+                    false
+                )
+            }
+            else -> {
+                throw UnsupportedOperationException("Retrieve is not supported for API type GEOCODING")
```

Comment:
It's better not to hardcode GEOCODING here, because `ApiType` might get extended

```suggestion
                throw UnsupportedOperationException("Retrieve is not supported for $apiType API type")
```